okhttp/src/main/java/okhttp3/Address.java

```diff
@@ -46,6 +52,8 @@
   final SSLSocketFactory sslSocketFactory;
   final HostnameVerifier hostnameVerifier;
   final CertificatePinner certificatePinner;
+  private Map<String, List<InetAddress>> cachedDnsResults =
```

Comment:
Happy for suggestions about this approach.  it effectively leaks memory as we retain the dns results.  I could clear them or find another way.